File: Ex03Widget2/app/src/main/java/com/tistory/firewallgogo/ex03widget2/MainActivity.java
package com.tistory.firewallgogo.ex03widget2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    //멤버변수
    TextView tv; //멤버변수 - 참조형, 기본값 null
    Button btn;
    EditText et;

    @Override
    protected void onCreate(Bundle savedInstanceState) {//onCreate가 끝나야 화면에 보인다.
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main); //xml 창에서 찾겠다.

        //xml에서 만들어진 view 객체를 찾아와서 참조변수에 대입
        tv =findViewById(R.id.tv); //tv = new Text(); 하지 않음 xml을 가져오기 때문에.  this. 생략함.
        btn=findViewById(R.id.btn); //this. 생략함.
        et=findViewById(R.id.et);

        //버튼이 클릭되면 TextView의 글씨를 변경!
        //버튼클릭 이벤트를 듣는 리스너 객체를 생성 및 붙이기
        btn.setOnClickListener(new View.OnClickListener() {
            //버튼이 클릭되면 자동으로 실행되는 콜백메소드
            @Override
            public void onClick(View view) {
                //EditText에 써 있는 글씨 얻어오기!
                Editable editable = et.getText(); //editable이 리턴값
                String str = editable.toString(); //가져온 값을 str로 옮긴다.

                //텍스트뷰의 글씨 변경하기
                tv.setText(str);

            }
        }); //addactionlistener 대신에 변경





    }
}

File: EX01HelloByJava/app/src/main/java/com/tistory/firewallgogo/ex01hellobyjava/MainActivity.kt
package com.tistory.firewallgogo.ex01hellobyjava

import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    var tv: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main);
//화면에 보여질 View객체 생성하기!!
        val btn = Button(this) //괄호 안에 무조건 this를 넣어야함 : 규칙.
        btn.text = "눌러주세요"
        //setContentView(btn);                   //add대신 set를 사용한다.
//글씨를 보여주는 View객체 생성
        tv = TextView(this)
        tv!!.text = "어서 눌러"
        tv!!.textSize = 32f
        tv!!.setTextColor(Color.BLUE)
        //setContentView(tv);                   //add는 여러개, set은 한개.
//activity는 한번에 하나만 구현가능 - btn 사라짐.
//btn 아래 깔린게 아니고 그냥 사라진 것.
//Activity는 한버넹 1개의 View만 설정할 수 있음.
//그래서 viewgroup을 1개 만들고 그 안에 View들을 추가
        val linearLayout = LinearLayout(this)
        linearLayout.orientation = LinearLayout.VERTICAL
        linearLayout.addView(btn)
        linearLayout.addView(tv)
        setContentView(linearLayout)
        //버튼 클릭하는 것에 반응하기 리스너추가하기
        btn.setOnClickListener {
            //텍스트뷰 글씨 변경하기
            tv!!.text = "Nice to meet you"
        }
    }
}